macros/src/main/scala/CostMetric.scala

```diff
@@ -113,18 +113,16 @@ object ExternalMetric extends CostMetricCompanion {
 /** The current default metric in barstools, re-defined by Donggyu. */
 // TODO: write tests for this function to make sure it selects the right things
 object DefaultMetric extends CostMetric with CostMetricCompanion {
-  override def cost(mem: Macro, lib: Macro): Option[BigInt] = {
+  override def cost(mem: Macro, lib: Macro): Option[Double] = {
     val memMask = mem.src.ports map (_.maskGran) find (_.isDefined) map (_.get)
     val libMask = lib.src.ports map (_.maskGran) find (_.isDefined) map (_.get)
     val memWidth = (memMask, libMask) match {
-      case (Some(1), Some(1)) | (None, _) => mem.src.width
+      case (Some(_), Some(1)) | (None, _) => mem.src.width
       case (Some(p), _) => p // assume that the memory consists of smaller chunks
     }
-    return Some(
-      (((mem.src.depth - 1) / lib.src.depth) + 1) *
-      (((memWidth - 1) / lib.src.width) + 1) *
-      (lib.src.depth * lib.src.width + 1) // weights on # cells
-    )
+    val depthCost = (mem.src.depth.toDouble / lib.src.depth.toDouble)
+    val widthCost = (memWidth.toDouble / lib.src.width.toDouble)
+    return Some(depthCost * widthCost)
```

Comment:
I know that we currently don't support using larger library SRAM macros than the requested ones, but this cost function would break if we did. Maybe add an assertion or TODO for now?

Reply:
Thats a good point. We actually do support larger library macros if we don't have a compiler.
I actually broke the old function for that use case. Will think about this a bit more.